app/src/components/domain/PrimarySectorSelectInput.tsx

```diff
@@ -0,0 +1,64 @@
+import { useState } from 'react';
+import {
+    SelectInput,
+    SelectInputProps,
+} from '@ifrc-go/ui';
+
+import type { GoApiResponse } from '#utils/restRequest';
+import { useRequest } from '#utils/restRequest';
+
+type PrimarySector = NonNullable<GoApiResponse<'/api/v2/primarysector'>>[number];
+export type Sector = Pick<PrimarySector, 'key' | 'label' | 'color' | 'is_deprecated'>;
+
+const keySelector = (d: Sector) => d.key;
+const labelSelector = (d: Sector) => d.label;
+
+type Props<NAME> = SelectInputProps<
+    number,
+    NAME,
+    PrimarySector,
+    any, // eslint-disable-line @typescript-eslint/no-explicit-any
+    'value' | 'name' | 'options' | 'keySelector' | 'labelSelector'
+> & {
+    className?: string;
+    name: NAME;
+    onChange: (newValue: number | undefined, name: NAME) => void;
+    value: number | undefined | null;
+    optionsFilter?: (item: PrimarySector) => boolean;
+}
+
+function PrimarySectorSelectInput<const NAME>(props: Props<NAME>) {
+    const {
+        name,
+        onChange,
+        value,
+        className,
+        ...otherProps
+    } = props;
+
+    const [opened] = useState(true);
+
+    const {
+        response: SectorOptions,
+    } = useRequest({
+        skip: !opened,
+        url: '/api/v2/primarysector',
+        preserveResponse: true,
+    });
+
+    return (
+        <SelectInput
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            {...otherProps}
```

Comment:
We don't need this as we will only be using the defined props. Also, let's disable the input when the request is pending.